Feat streaming api (#20)
Pure-Farming/Documentation: docs/streaming-api/datasets.md

---

> title: Datasets Endpoint

Reviewer: ```suggestion
title: Datasets
```

        "updates": "https://api.dev.purefarming.com/data/streaming/datasets/holdings/updates",
        "containedTypes": [
          "/core/holding"
        ]
    },
    ...
]
```

This response is a list of [dataset response](#dataset-response)s

# Dataset Endpoint

The dataset endpoint returns metadata about a single dataset

```
GET /data/streaming/datasets/{datasetid}
```
where `datasetId` is the name of the dataset

## Dataset Response

```json
{
    "name": "holdings",
    "url": "https://api.dev.purefarming.com/data/streaming/datasets/holdings",
    "changes": "https://api.dev.purefarming.com/data/streaming/datasets/holdings/changes",
    "updates": "https://api.dev.purefarming.com/data/streaming/datasets/holdings/updates",
    "containedTypes": [
      "/core/holding"
    ]
}
```

| Response Item | Description | Data Type |
| ------------- | ----------- | --------- |
| **Name** | The name of the dataset (also called the dataset id) | string | 
| **URL** | A link to the dataset metadata | URI | 
| **Changes** | A link to get the latest changes to the resources within the dataset | URI |
| **Updates** | A link publish updates for a resource | URI |

> | **Contained Types** |  A list of resource types contained in this dataset | List of strings |

Reviewer: ```suggestion
| **Contained Types** |  An array of resource types contained in this dataset | Array of Strings |
```
